### PR TITLE
8314212: Crash when loading cnn.com in WebView

### PR DESCRIPTION
**Pull request by jaybhaskar**
Issue: WCMediaPlayer does not support HTTP live streaming m3u8 format. Giving error com.sun.javafx.webkit.prism.WCMediaPlayerImpl onError WARNING: onError, errCode=0, msg=Unsupported protocol "data"

Solution: Do not invoke the current media time operation in an error state. As the WCMediaPlayer might not be a valid object.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issue
 * [JDK-8314212](https://bugs.openjdk.org/browse/JDK-8314212): Crash when loading cnn.com in WebView (**Bug** - P2)

### Reviewers
 * [Kevin Rushforth](https://openjdk.org/census#kcr) (@kevinrushforth - **Reviewer**)
 * [Hima Bindu Meda](https://openjdk.org/census#hmeda) (@HimaBinduMeda - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx.git pull/1212/head:pull/1212` \
`$ git checkout pull/1212`

Update a local copy of the PR: \
`$ git checkout pull/1212` \
`$ git pull https://git.openjdk.org/jfx.git pull/1212/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1212`

View PR using the GUI difftool: \
`$ git pr show -t 1212`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/1212.diff">https://git.openjdk.org/jfx/pull/1212.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jfx/pull/1212#issuecomment-1681817928)

### CODE REVIEW

**Comment by kevinrushforth** [comment]
@jaybhaskar I changed the JBS bug title to something more reflective of the problem, so you'll need to edit this PR title to match.

/reviewers 2

File: modules/javafx.web/src/main/native/Source/WebCore/platform/graphics/java/MediaPlayerPrivateJava.cpp [hunk 1]
```diff
@@ -430,6 +430,12 @@ float MediaPlayerPrivate::currentTime() const
         LOG_TRACE1("MediaPlayerPrivate currentTime returns (seekTime): %f\n", m_seekTime);
         return m_seekTime;
     }
+
+    // in case of hls media m3u8 format check network state
+    // since jfx media do not support hls live streaming protocol
+    if ( MediaPlayerNetworkState::NetworkError == MediaPlayer::NetworkState::NetworkError)
```
Comment (kevinrushforth) [inline modules/javafx.web/src/main/native/Source/WebCore/platform/graphics/java/MediaPlayerPrivateJava.cpp:436]: Are there other errors that might occur, or is this the only one you need to check?

Minor: delete the extra space after the `(`.

File: modules/javafx.web/src/main/native/Source/WebCore/rendering/RenderLayerCompositor.cpp [hunk 2]
```diff
@@ -1196,6 +1196,11 @@ void RenderLayerCompositor::computeCompositingRequirements(RenderLayer* ancestor
 #endif
     }
 
+// in case of video data format like m3u8 error case ,check if layer needs compositing
+#if PLATFORM(JAVA)
+    if(willBeComposited != needsToBeComposited(layer, queryData))
```
Comment (kevinrushforth) [inline modules/javafx.web/src/main/native/Source/WebCore/rendering/RenderLayerCompositor.cpp:1201]: Minor: add extra space after `if`.

**Review by kevinrushforth — COMMENTED** [review]
I left one question about the fix (and also noted a couple minor formatting issues).

I presume you've tested this on all platforms both with media that produces the error condition, and also verifying that valid media continues to play correctly?

**Comment by kevinrushforth** [comment]
@HimaBinduMeda Can you be the second reviewer?

File: modules/javafx.web/src/main/native/Source/WebCore/platform/graphics/java/MediaPlayerPrivateJava.cpp (hunk 1 above)
Comment (jaybhaskar) [reply modules/javafx.web/src/main/native/Source/WebCore/platform/graphics/java/MediaPlayerPrivateJava.cpp]: done

File: modules/javafx.web/src/main/native/Source/WebCore/rendering/RenderLayerCompositor.cpp (hunk 2 above)
Comment (jaybhaskar) [reply modules/javafx.web/src/main/native/Source/WebCore/rendering/RenderLayerCompositor.cpp]: done

**Comment by jaybhaskar** [comment]
I have tested on mac/Linux , window test is ongoing

File: modules/javafx.web/src/main/native/Source/WebCore/platform/graphics/java/MediaPlayerPrivateJava.cpp (hunk 1 above)
Comment (kevinrushforth) [reply modules/javafx.web/src/main/native/Source/WebCore/platform/graphics/java/MediaPlayerPrivateJava.cpp]: The answer to my question is still pending. Please take a look at whether or not this is the only error you need to check. If it is, can you say why?

Comment (jaybhaskar) [reply modules/javafx.web/src/main/native/Source/WebCore/platform/graphics/java/MediaPlayerPrivateJava.cpp]: Once the Player is in an error state, play, pause, seek, etc. all of these operations would not be called. Only the current time which is the last played point of time needs to be updated to the player user interface.

Comment (kevinrushforth) [reply modules/javafx.web/src/main/native/Source/WebCore/platform/graphics/java/MediaPlayerPrivateJava.cpp]: OK. My question then is whether `MediaPlayerNetworkState::NetworkError` is the only error condition you need to consider.

**Review by HimaBinduMeda — APPROVED** [review]
Changes look fine. Testing looks good.

**Review by kevinrushforth — APPROVED** [review]
Code changes look good. All my testing is green.

**Pull request closed without merging**